maintenance/migratestatus/migrator/migrator.go

```diff
@@ -202,49 +199,52 @@ type Migrator struct {
 	repo            string
 	continueOnError bool
 
-	client *ghclient.Client
+	client *github.Client
 	Mode
 }
 
-// New creates a new migrator with specified options.
-//
-// If dryRun is true it will only perform GET requests that do not change github.
-func New(mode Mode, token, org, repo string, dryRun, continueOnError bool) *Migrator {
+// New creates a new migrator with specified options and client.
+func New(mode Mode, client *github.Client, org, repo string, continueOnError bool) *Migrator {
 	return &Migrator{
 		org:             org,
 		repo:            repo,
 		continueOnError: continueOnError,
-		client:          ghclient.NewClient(token, dryRun),
+		client:          client,
 		Mode:            mode,
 	}
 }
 
-func (m *Migrator) processPR(pr *github.PullRequest) error {
-	if pr == nil {
-		return fmt.Errorf("migrator cannot process a nil PullRequest")
-	}
-	if pr.Head == nil {
-		return fmt.Errorf("migrator cannot process a PullRequest with a nil 'Head' field")
-	}
-	if pr.Head.SHA == nil {
-		return fmt.Errorf("migrator cannot process a PullRequest with a nil 'Head.SHA' field")
-	}
-
-	combined, err := m.client.GetCombinedStatus(m.org, m.repo, *pr.Head.SHA)
+func (m *Migrator) processPR(pr github.PullRequest) error {
+	combined, err := m.client.GetCombinedStatus(m.org, m.repo, pr.Head.SHA)
 	if err != nil {
 		return err
 	}
 	actions := m.processStatuses(combined)
 
 	for _, action := range actions {
-		if _, err = m.client.CreateStatus(m.org, m.repo, *pr.Head.SHA, action); err != nil {
+		if err := m.client.CreateStatus(m.org, m.repo, pr.Head.SHA, action); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // Migrate will retire/migrate/copy statuses for all matching PRs.
-func (m *Migrator) Migrate(prOptions *github.PullRequestListOptions) error {
-	return m.client.ForEachPR(m.org, m.repo, prOptions, m.continueOnError, m.processPR)
+func (m *Migrator) Migrate() error {
+	prs, err := m.client.GetPullRequests(m.org, m.repo)
```

Comment:
This will get the full list of all PRs before it starts processing unlike `ForEachPR` which process a page at a time. It is nice to run the tool in dry run first and see how it behaves on a few PRs, but this change would make that very expensive to do.
We should also ensure that we are only listing open PRs.
https://github.com/kubernetes/test-infra/blob/e38b06602359afa6531c47abcac10dfa371063b1/pkg/ghclient/wrappers.go#L103-L142

Reply:
Yeah, I figured that might be contentious -- I can change to do open ones only

Reply:
:+1: Listing PRs before starting to process should be fine if we are only listing open PRs (for k/k that would cost ~10 tokens instead of ~430 for all PRs).

Reply:
Seems like `state:open` is the default for this API. Do we need to change anything else here?

Reply:
For kubernetes, the difference between the paginated approach and getting all of them is 8 tokens right now. Seems like something that isn't super important?

Reply:
Yep, if the default is `state:open` then this isn't a concern.